app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        return render_invalid_filter_note_type unless valid_note_type?
+        render json: notes, status: :ok, each_serializer: IndexNoteSerializer
+      end
+
+      def show
+        render json: note, status: :ok, serializer: ShowNoteSerializer
+      end
+
+      private
+
+      def filtering_params
+        params.permit [:note_type]
+      end
+
+      def note_type
+        params[:note_type]
+      end
+
+      def valid_note_type?
+        note_type.nil? || Note.note_types.keys.include?(params[:note_type])
+      end
```

Comment:
esto ya es note_type
```suggestion
        note_type.nil? || Note.note_types.keys.include?(note_type)
      end
```